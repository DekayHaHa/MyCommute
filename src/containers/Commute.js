import React from 'react';

export const Commute = () => {
  return (
    <div>

    </div>
  )
}