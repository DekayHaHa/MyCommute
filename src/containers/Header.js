import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { signUserOut } from '../thunks/signUserOut'
import UserNavBar from './UserNavBar'
import moment from 'moment';
import '../styles/header.css'

export class Header extends Component {

  render() {
    const { user, currentWeather, signUserOut } = this.props
    let time = currentWeather ? moment.unix(currentWeather.time).format('LT') : null;
    return (
      <div className='main-header'>
        <div className='header'>
          <h1 className='title'>MyCommute</h1>
          <div className='current-weather'>
            {!user.id && <Link to='/login'>LOGIN USER</Link>}
            {currentWeather &&
              <div>
                <h4>Hello, {user.userName} <Link to='/'><button className='signout-btn' onClick={() => signUserOut()}>SignOut</button></Link></h4>
                <p>Current as of {time}</p>
                <p>{currentWeather.summary}</p>
                <p>{currentWeather.temperature} FÂ°</p>
              </div>}
          </div >
        </div>
        <Route path={`/user/${user.id}`} component={UserNavBar} />
      </div>
    );
  }
}

Header.propTypes = {
  user: PropTypes.object,
  signUserOut: PropTypes.func,
  currentWeather: PropTypes.object
};

export const mapDispatchToProps = (dispatch) => ({
  signUserOut: () => dispatch(signUserOut())
});

export const mapStateToProps = store => ({
  user: store.user,
  currentWeather: store.weather.current
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);