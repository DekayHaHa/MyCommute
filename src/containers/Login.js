import React, { Component } from 'react';
// import { Route, Switch } from 'react-router-dom';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getUser } from '../thunks/getUser'

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      userName: '',
      password: ''
    }
  }
  handleSubmit = (e) => {
    e.preventDefault()
    const { userName, password } = this.state
    this.props.getUser(userName, password)
    this.setState({ userName: '', password: '' })
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }

  render() {
    const { user, error } = this.props
    const { userName, password } = this.state
    return (
      <div>
        {user.id && <Redirect to={`/user/${user.id}/commutes`} />}
        <form onSubmit={this.handleSubmit}>
          <input className='input-button'
            name='userName'
            type='text'
            value={userName}
            placeholder='USERNAME...'
            onChange={this.handleChange}
          />
          <input className='input-button'
            name='password'
            type='password'
            value={password}
            placeholder='PASSWORD...'
            onChange={this.handleChange}
          />
          <button className='form-button' onClick={this.fetchData}>Sign In</button>
          {error && <p className='input-error' >{error}</p>}
        </form>
      </div>
    );
  }
}

Login.propTypes = {
  user: PropTypes.object,
  getUser: PropTypes.func,
  error: PropTypes.string
};

export const mapDispatchToProps = (dispatch) => ({
  getUser: (name, pass) => dispatch(getUser(name, pass))
});

export const mapStateToProps = store => ({
  user: store.user,
  error: store.error
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);