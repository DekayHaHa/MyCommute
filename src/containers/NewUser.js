import React, { Component } from 'react';
// import { Route, Switch } from 'react-router-dom';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createUser } from '../thunks/createUser'

export class NewUser extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      userName: '',
      password: '',
      verifyPassword: '',
      zip: 0,
      lat: 0,
      lng: 0,
      locCheck: false,
    }
  }

  componentDidMount() {
    let answer = window.confirm('Can we use your web browesr location?')
    if (answer) {
      this.setState({ locCheck: true })
    }
  }

  handleSubmit = (e) => {
    e.preventDefault()
    console.log(this.state)
    const { userName, password, zip, lat, lng, email } = this.state
    let user = {
      email,
      userName,
      password,
      zip,
      lat,
      lng
    }
    this.props.createUser(user)
    this.setState({ userName: '', password: '', verifyPassword: '', email: '', zip: 0, lat: 0, lng: 0 })
  }

  successCallback = (pos) => {
    this.setState({ lat: pos.coords.latitude, lng: pos.coords.longitude })
  }

  errorCallback = (err) => {
    console.log('in err', err)
    return err
  }

  getUserLocation = () => {
    const opt = { timeout: 10000 }
    navigator.geolocation.getCurrentPosition(this.successCallback, this.errorCallback, opt);
    console.log(this.state)
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }

  render() {
    const { user, error } = this.props
    const { userName, password, verifyPassword, email, zip, lat, lng, locCheck } = this.state
    if (!lat && !lng && locCheck) this.getUserLocation()
    return (
      <div>
        {user.id && <Redirect to='/' />}
        <form onSubmit={this.handleSubmit}>
          <input
            name='email'
            type='text'
            value={email}
            placeholder='EMAIL...'
            onChange={this.handleChange}
          />
          <input
            name='userName'
            type='text'
            value={userName}
            placeholder='USERNAME...'
            onChange={this.handleChange}
          />
          <input
            name='zip'
            type='number'
            value={zip}
            placeholder='ZIP CODE...'
            onChange={this.handleChange}
          />
          {zip.length > 0 && zip.length !== 5 && <p>Please enter a 5 digit Zip Code.</p>}
          <input
            name='password'
            type='password'
            value={password}
            placeholder='PASSWORD...'
            onChange={this.handleChange}
          />
          <input
            name='verifyPassword'
            type='password'
            value={verifyPassword}
            placeholder='VERIFY PASSWORD...'
            onChange={this.handleChange}
          />
          {verifyPassword !== password && <p>Passwords Do Not Match</p>}
          {error && <p>{error}</p>}
          <button onClick={this.fetchData}>Sign Up</button>
        </form>
      </div>
    );
  }
}

NewUser.propTypes = {
  user: PropTypes.object,
  createUser: PropTypes.func,
  error: PropTypes.string
};

export const mapDispatchToProps = (dispatch) => ({
  createUser: (user) => dispatch(createUser(user))
});

export const mapStateToProps = store => ({
  user: store.user,
  error: store.error
});

export default connect(mapStateToProps, mapDispatchToProps)(NewUser);