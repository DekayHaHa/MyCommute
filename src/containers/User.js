import React, { Component } from 'react';
// import { Route, Switch } from 'react-router-dom';
import { Link, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getUser } from '../thunks/getUser'

export class User extends Component {


  render() {
    return (
      <nav>

      </nav>
    );
  }
}

User.propTypes = {
  user: PropTypes.object,
  signinUser: PropTypes.func
};

export const mapDispatchToProps = (dispatch) => ({
  getUser: (name, pass) => dispatch(getUser(name, pass))
});

export const mapStateToProps = store => ({
  user: store.user,
  error: store.error
});

export default connect(mapStateToProps, mapDispatchToProps)(User);