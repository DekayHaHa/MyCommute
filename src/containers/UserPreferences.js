import React from 'react';

export const UserPreferences = () => {
  return (
    <div>
      <h3>I'm Preferences</h3>
    </div>
  )
}