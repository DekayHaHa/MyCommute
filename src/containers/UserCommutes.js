import React, { Component } from 'react';
// import { Route, Switch } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Commute } from '../components/Commute'

export class UserCommutes extends Component {

  checkPref = () => {
    const { commutes, pref, weather } = this.props
    const commutesToRender = []
    let recommendation = { pri: 10 }
    Object.keys(commutes).forEach(commKey => {
      if (commKey === 'id') return
      Object.keys(pref).forEach(prefKey => {
        if (prefKey === 'id') return
        if (pref[prefKey].temp < weather.current.temperature && recommendation.pri > pref[prefKey].priority) {
          recommendation.pri = pref[prefKey].priority
          recommendation['recom'] = prefKey
          recommendation['price'] = pref[prefKey].price
          recommendation['estTravel'] = pref[prefKey].estimatedTime
        }
      }, {})
      commutesToRender.push({
        ...recommendation,
        title: commKey
      })
    })
    // const currentTime = moment().format('LT')
    // const timeLeft = moment.duration(moment(currentTime, 'HHmm').diff(moment('1800', 'HH:mm'))).humanize()
    // console.log(currentTime, timeLeft)
    return commutesToRender.map(commute => <Commute key={commute.title} {...commute} />)
  }

  render() {
    const { weather } = this.props
    return (
      <div className='container'>
        {weather.today && this.checkPref()}
      </div>
    );
  }
}

UserCommutes.propTypes = {
  user: PropTypes.object,
  error: PropTypes.string,
  pref: PropTypes.object,
  commutes: PropTypes.object,
  weather: PropTypes.object
};

export const mapStateToProps = store => ({
  user: store.user,
  error: store.error,
  pref: store.pref,
  commutes: store.commutes,
  weather: store.weather
});

export default connect(mapStateToProps)(UserCommutes);