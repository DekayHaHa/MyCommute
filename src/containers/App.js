import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import Login from './Login'
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Header from './Header'
// import NewUser from './NewUser'
import UserCommutes from './UserCommutes'
import UserWeather from './UserWeather'
import UserPreferences from './UserPreferences'
import { Home } from '../components/Home'
import { PageError } from '../components/PageError'
import '../styles/App.css'

export class App extends Component {

  render() {
    const { user } = this.props
    return (
      <div className="App">
        {/* {user.id && this.getUserWeather()} */}
        <Route to='/' component={Header} />
        <Switch>
          <Route exact path='/' component={Home} />
          <Route exact path='/login' component={Login} />
          <Route exact path={`/user/${user.id}/commutes`} component={UserCommutes} />
          <Route exact path={`/user/${user.id}/preferences`} component={UserPreferences} />
          <Route exact path={`/user/${user.id}/weather`} component={UserWeather} />
          {/* <Route exact path='/newUser' component={NewUser} /> */}
          <Route component={PageError} />
        </Switch>
      </div>
    );
  }
}

App.propTypes = {
  user: PropTypes.object
};

export const mapStateToProps = store => ({
  user: store.user
});

export default connect(mapStateToProps)(App);