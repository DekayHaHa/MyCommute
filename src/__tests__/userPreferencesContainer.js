import React from 'react'
import { UserPreferences, mapStateToProps } from '../containers/UserPreferences'
import { shallow } from 'enzyme'

describe('UserPreferences', () => {
  let wrapper;
  const mockUser = { id: 1, userName: 'mahk' }
  const mockPref = { id: 1, tram: { temp: 80 } }
  beforeEach(() => {
    wrapper = shallow(<UserPreferences pref={mockPref} user={mockUser} />)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('map props', () => {
    it('should map state to props', () => {
      const mockState = {
        user: { id: 1, userName: 'mahk' },
        error: 'im wrong',
        pref: { id: 1 },
        commutes: { id: 1 },
        notMe: 'shouldnt show'
      }
      const expectedState = {
        user: { id: 1, userName: 'mahk' },
        error: 'im wrong',
        pref: { id: 1 },
        commutes: { id: 1 },
      }
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expectedState);
    })
  })
})