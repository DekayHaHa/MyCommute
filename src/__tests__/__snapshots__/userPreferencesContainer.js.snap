// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPreferences should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserPreferences
    pref={
      Object {
        "id": 1,
        "tram": Object {
          "temp": 80,
        },
      }
    }
    user={
      Object {
        "id": 1,
        "userName": "mahk",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Pref
          temp={80}
          title="tram"
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "tram",
        "nodeType": "function",
        "props": Object {
          "estimatedTime": undefined,
          "price": undefined,
          "priority": undefined,
          "temp": 80,
          "title": "tram",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Pref
            temp={80}
            title="tram"
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "tram",
          "nodeType": "function",
          "props": Object {
            "estimatedTime": undefined,
            "price": undefined,
            "priority": undefined,
            "temp": 80,
            "title": "tram",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
